crypto/libsodium-fork/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c

```diff
@@ -2583,3 +2692,17 @@ ge25519_from_uniform(unsigned char s[32], const unsigned char r[32])
 
     ge25519_p3_tobytes(s, &p3);
 }
+
+int
+sc25519_is_canonical_vartime(const unsigned char s[32])
+{
+    if ((s[31] & 240) == 0)
```

Comment:
This is the same thing as https://github.com/jedisct1/libsodium/commit/61368716075992568a0adcfd696359e206827017#diff-4db2d523f04cf0c485674b4e3a5f4f15df23d9bc4f33a6d4b4d701402912141d right?

Reply:
yeah
if think ours is clearer